core/src/main/java/org/geysermc/geyser/extension/GeyserExtensionLoader.java

```diff
@@ -164,7 +164,69 @@ protected void loadAllExtensions(@NonNull ExtensionManager extensionManager) {
             Map<String, GeyserExtensionContainer> loadedExtensions = new LinkedHashMap<>();
 
             Pattern[] extensionFilters = this.extensionFilters();
-            List<Path> extensionPaths = Files.walk(extensionsDirectory).toList();
+
+            Path updateDirectory = extensionsDirectory.resolve("update");
+            List<Path> extensionPaths;
+            if (Files.isDirectory(updateDirectory)) {
+                // Get the current extensions and store them in a map
+                Map<String, Path> extensionFiles = new HashMap<>();
+                extensionPaths = Files.list(extensionsDirectory).toList();
+                extensionPaths.forEach(path -> {
+                    if (Files.isDirectory(path)) {
+                        return;
+                    }
+
+                    // Only look at files that meet the extension filter
+                    for (Pattern filter : extensionFilters) {
+                        if (!filter.matcher(path.getFileName().toString()).matches()) {
+                            return;
+                        }
+                    }
+
+                    try {
+                        // Try load the description, so we know it's a valid extension
+                        GeyserExtensionDescription description = this.extensionDescription(path);
```

Comment:
this code is duplicated three times, can it be turned into a function that takes a directory Path and some sort of consumer for each description and other necessary values